flank-scripts/src/main/kotlin/flank/scripts/github/GithubApi.kt

```diff
@@ -11,49 +12,103 @@ import com.jcabi.github.Releases
-import flank.scripts.exceptions.toGithubException
+import flank.scripts.exceptions.mapErrorToGithubException
+import flank.scripts.github.objects.GitHubCommitListDeserializer
+import flank.scripts.github.objects.GitHubCreateIssueCommentRequest
+import flank.scripts.github.objects.GitHubCreateIssueCommentResponseDeserializer
+import flank.scripts.github.objects.GitHubCreateIssueRequest
+import flank.scripts.github.objects.GitHubCreateIssueResponseDeserializer
+import flank.scripts.github.objects.GitHubUpdateIssueRequest
+import flank.scripts.github.objects.GithubPullRequestDeserializer
+import flank.scripts.github.objects.GithubPullRequestListDeserializer
+import flank.scripts.github.objects.GithubWorkflowRunsSummaryDeserializer
+import flank.scripts.utils.toJson
 
-// ============= HTTP GITHUB API =============
+private const val URL_BASE = "https://api.github.com/repos/Flank/flank"
 
+// ============= HTTP GITHUB API =============
+// GET
 suspend fun getPrDetailsByCommit(commitSha: String, githubToken: String) =
-    Fuel.get("https://api.github.com/repos/flank/flank/commits/$commitSha/pulls")
+    Fuel.get("$URL_BASE/commits/$commitSha/pulls")
         .appendGitHubHeaders(githubToken, "application/vnd.github.groot-preview+json")
         .awaitResult(GithubPullRequestListDeserializer)
-        .mapClientError { it.toGithubException() }
+        .mapErrorToGithubException()
         .onError { println("Could not download info for commit $commitSha, because of ${it.message}") }
 
 suspend fun getLatestReleaseTag(githubToken: String) =
-    Fuel.get("https://api.github.com/repos/flank/flank/releases/latest")
+    Fuel.get("$URL_BASE/releases/latest")
         .appendGitHubHeaders(githubToken)
         .awaitResult(GithubReleaseDeserializable)
-        .mapClientError { it.toGithubException() }
-
-fun deleteOldTag(tag: String, username: String, password: String) =
-    Fuel.delete(DELETE_ENDPOINT + tag)
-        .authentication()
-        .basic(username, password)
-        .response()
-        .third
-        .mapClientError { it.toGithubException() }
+        .mapErrorToGithubException()
 
 suspend fun getGitHubPullRequest(githubToken: String, issueNumber: Int) =
-    Fuel.get("https://api.github.com/repos/Flank/flank/pulls/$issueNumber")
+    Fuel.get("$URL_BASE/pulls/$issueNumber")
         .appendGitHubHeaders(githubToken)
         .awaitResult(GithubPullRequestDeserializer)
-        .mapClientError { it.toGithubException() }
+        .mapErrorToGithubException()
 
 suspend fun getGitHubIssue(githubToken: String, issueNumber: Int) =
-    Fuel.get("https://api.github.com/repos/Flank/flank/issues/$issueNumber")
+    Fuel.get("$URL_BASE/issues/$issueNumber")
         .appendGitHubHeaders(githubToken)
         .awaitResult(GithubPullRequestDeserializer)
-        .mapClientError { it.toGithubException() }
+        .mapErrorToGithubException()
+
+suspend fun getGitHubIssueList(githubToken: String, parameters: Parameters = emptyList()) =
```

Comment:
Pls, expose the return type for convenient reading. Also for the rest added functions.

Reply:
Good point, thanks for suggestion